src/main/java/com/code_intelligence/jazzer/mutation/mutator/collection/MapMutatorFactory.java

```diff
@@ -0,0 +1,254 @@
+      for (Map.Entry<K, V> entry : map.entrySet()) {
+        keyMutator.write(entry.getKey(), out);
+        valueMutator.write(entry.getValue(), out);
+      }
+    }
+
+    @Override
+    protected Map<K, V> makeDefaultInstance() {
+      // Wrap in an immutable view for additional protection against accidental mutation in fuzz
+      // tests.
+      return toImmutableMapView(new HashMap<>(maxInitialSize()));
+    }
+
+    @Override
+    public void initInPlace(Map<K, V> reference, PseudoRandom prng) {
+      int targetSize = prng.closedRange(minInitialSize(), maxInitialSize());
+      Map<K, V> map = underlyingMutableList(reference);
+      map.clear();
+      for (int i = 0; i < targetSize; i++) {
+        initElement(map, prng);
+      }
+    }
+
+    @Override
+    public void mutateInPlace(Map<K, V> reference, PseudoRandom prng) {
+      Map<K, V> map = underlyingMutableList(reference);
+      if (map.isEmpty()) {
+        initElement(map, prng);
+      } else if (!prng.trueInOneOutOf(4)) {
+        // Choose a random entry, and then key or value, to mutate.
+        int i = prng.indexIn(map.size());
+        map.entrySet().stream().skip(i).findFirst().ifPresent(
+            entry -> mutateElement(entry, map, prng));
+      } else {
+        // This will increase the map size up to max and then oscillate around it.
+        // Smaller sizes are already used, so increasing the size should be a valid choice.
+        // FIXME: Think about a better strategy to change the size.
+        int currentSize = map.size();
+        if (currentSize < maxSize) {
+          // Create a new entry, as a deep copy could be expensive.
+          initElement(map, prng);
+        } else if (currentSize > minSize) {
+          // Remove a random entry.
+          int i = prng.indexIn(currentSize);
+          map.keySet().stream().skip(i).findFirst().ifPresent(map::remove);
+        } else {
+          // One element map, mutate that one.
+          mutateElement(map.entrySet().iterator().next(), map, prng);
+        }
+      }
+    }
+
+    private void initElement(Map<K, V> map, PseudoRandom prng) {
+      // Lookup an unused key first, as the map may already contain and
+      // deduplicate it. Keys are solely based on the prng so that an
+      // unused one should eventually be found. In case all or most values
+      // of the given type are already used, give up after some attempts.
+      int attempts = 0;
+      K key;
+      do {
+        if (attempts++ > DEFAULT_INIT_ATTEMPTS) {
+          // Give up, as no unused key could be found.
+          return;
+        }
+        key = keyMutator.init(prng);
+      } while (map.containsKey(key));
+      map.put(key, valueMutator.init(prng));
+    }
+
+    private void mutateElement(Map.Entry<K, V> entry, Map<K, V> map, PseudoRandom prng) {
+      K key = entry.getKey();
+      V value = entry.getValue();
+      map.remove(key);
+      if (prng.choice()) {
+        key = keyMutator.mutate(key, prng);
```

Comment:
This also needs a retry as you may mutate the key to be something that is already in the map.

Reply:
Good catch! Changed as well. If no unused key can be found during mutation, the value is mutated instead.